app/components/wallet/trezorConnect/ConnectDialog.js

```diff
@@ -0,0 +1,141 @@
+
+import ProgressStepBlock from './common/ProgressStepBlock';
+import HelpLinkBlock from './common/HelpLinkBlock';
+import ErrorBlock from './common/ErrorBlock';
+
+import connectLoadGIF from '../../../assets/images/trezor/connect-load.gif';
+import connectStartGIF from '../../../assets/images/trezor/connect-start.gif';
+import connectErrorSVG from '../../../assets/images/trezor/connect-error.inline.svg';
+
+import type { ProgressInfo } from '../../../stores/ada/TrezorConnectStore';
+import { StepStateOption } from '../../../stores/ada/TrezorConnectStore';
+
+import styles from './ConnectDialog.scss';
+
+const messages = defineMessages({
+  title: {
+    id: 'wallet.trezor.dialog.title.label',
+    defaultMessage: '!!!Connect to Trezor Hardware Wallet',
+    description: 'Label "Connect to Trezor Hardware Wallet" on the Connect to Trezor Hardware Wallet dialog.'
+  },
+  connectIntroTextLine1: {
+    id: 'wallet.trezor.dialog.trezor.step.connect.introText.line.1',
+    defaultMessage: '!!!After connecting your Trezor device to the computer press the Connect button.',
+    description: 'Header text of about step on the Connect to Trezor Hardware Wallet dialog.'
+  },
+  connectIntroTextLine2: {
+    id: 'wallet.trezor.dialog.trezor.step.connect.introText.line.2',
+    defaultMessage: '!!!A new tab will appear, please follow the instructions in the new tab.',
+    description: 'Header text of about step on the Connect to Trezor Hardware Wallet dialog.'
+  },
+  connectIntroTextLine3: {
+    id: 'wallet.trezor.dialog.trezor.step.connect.introText.line.3',
+    defaultMessage: '!!!This process shares the Cardano public key with Yoroi.',
+    description: 'Header text of about step on the Connect to Trezor Hardware Wallet dialog.'
+  },
+  connectButtonLabel: {
+    id: 'wallet.trezor.dialog.trezor.connect.button.label',
+    defaultMessage: '!!!Connect',
+    description: 'Label for the "Connect" button on the Connect to Trezor Hardware Wallet dialog.'
+  },
+});
+
+messages.fieldIsRequired = globalMessages.fieldIsRequired;
+
+type Props = {
+  progressInfo: ProgressInfo,
+  isActionProcessing: boolean,
+  error: ?LocalizableError,
+  goBack: Function,
+  submit: Function,
+  cancel: Function,
+};
+
+@observer
+export default class ConnectDialog extends Component<Props> {
+
+  static contextTypes = {
+    intl: intlShape.isRequired
+  };
+
+  render() {
+    const { intl } = this.context;
+
+    // introBlock
+    const introBlock = (
+      <div className={styles.headerBlock}>
+        <span>{intl.formatMessage(messages.connectIntroTextLine1)}</span><br />
+        <span>{intl.formatMessage(messages.connectIntroTextLine2)}</span><br />
+        <span>{intl.formatMessage(messages.connectIntroTextLine3)}</span><br />
+      </div>);
+
+    // middleBlock + backButton selection depending upon state
+    let middleBlock = null;
+    let backButton = null;
+
+    switch (this.props.progressInfo.stepState) {
+      case StepStateOption.LOAD:
```

Comment:
I would delete the Option from `StepStateOption` so it just becomes `StepState`.

Reply:
okay

Reply:
why are we not reusing that? the states should be the same?

Reply:
export type ProgressStepEnum = 0 | 1 | 2;
export const ProgressStep = {
  ABOUT: 0,
  CONNECT: 1,
  SAVE: 2,
};

do like this

Reply:
DONE